### PR TITLE
Update NodeJS to v16.17.0

### PR DESCRIPTION
**Pull request by noahtallen**
### Proposed Changes

Update NodeJs to v16.17.0 from v16.13.2. Previous update: #60120.

I was recently reminded that we haven't done a NodeJS update in Calypso for a few months when I was working in another repo. This brings us up-to-date with the latest stable LTS.

Changelog: https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V16.md#16.17.0

Notable changes:
- 2 security updates related to OpenSSL
- Importing JSON modules now requires experimental import assertions syntax
- Adds experimental fetch API
- Adds experimental command-line argument parser API
- Adds experimental ESM Loader Hooks API
- Adds experimental test runner

### Testing Instructions
- [x] All CI should pass
- [x] Run `nvm install && nvm use` (or `volta install node@16.17.0`) and test local development workflows
- [x] Run the "base image build" with a test tag and verify it works correctly.

I also ran CI manually like so:
1. I ran a [base image build](https://teamcity.a8c.com/buildConfiguration/calypso_BuildBaseImages/8501625) using this branch with a custom docker image tag `test-node-16-17-0`
2. I ran a normal [docker image build](https://teamcity.a8c.com/buildConfiguration/calypso_BuildDockerImage/8501765) utilizing that base image by overriding the docker tag in the build parameters.
3. I ran both e2e suites ([desktop](https://teamcity.a8c.com/buildConfiguration/calypso_Calypso_E2E_Playwright_desktop/8501775), [mobile](https://teamcity.a8c.com/buildConfiguration/calypso_Calypso_E2E_Playwright_mobile/8501777)), again using the new base image.
4. I verified that all builds were successful and that the calypso.live image works properly.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -142,7 +142,7 @@ commands:
 jobs:
   wp-desktop-assets:
     docker:
-      - image: circleci/node:12.16.2-browsers
+      - image: circleci/node:16.17.0-browsers
```
Comment (noahtallen) [inline .circleci/config.yml:145]: While this appears to be a big update, the workflow below already uses `nvm` to update Node to a more recent version, so it wasn't using 12.16 anyways.

**Review by worldomonation — APPROVED** [review]
While I'm not an expert on toolchain changes like this one, I've verified on the E2E builds that `node 16.17.0` is now being used:

```
Now using node v16.17.0 (npm v8.15.0)
```

File: .circleci/config.yml (hunk 1 above)
Comment (fullofcaffeine) [reply .circleci/config.yml:145]: Could you elaborate?

Also, do you know why this is in a `.circleci` directory? Do we still use CircleCI?

Comment (noahtallen) [reply .circleci/config.yml:145]: Yeah, as far as I know, we use circleci to build the desktop app. (Not the web app)

Comment (worldomonation) [reply .circleci/config.yml:145]: To add to that, I had a brief chat with @nsakaimbo some months ago about possibly moving Desktop builds (not e2e) to TeamCity. However, as Desktop builds require also generate macOS and Windows binaries we decided the effort was not worth the results.

Comment (noahtallen) [reply .circleci/config.yml:145]: Yeah, I doubt we have Windows or macOS server environments in TeamCity 😅

Comment (fullofcaffeine) [reply .circleci/config.yml:145]: Alright, thanks for clarifying folks!

**Review by fullofcaffeine — APPROVED** [review]

**Pull request merged**